examples/cloud-operations/workload-identity-federation/azure.tf

```diff
@@ -0,0 +1,177 @@
+locals {
+  username    = "azureuser"
+  app_name    = "test-app"
+  custom_data = <<CUSTOM_DATA
+#!/bin/bash
+cat <<EOF > /home/${local.username}/credentials.json
+{
+  "type": "external_account",
+  "audience": "//iam.googleapis.com/projects/${module.prj.number}/locations/global/workloadIdentityPools/${google_iam_workload_identity_pool.pool.workload_identity_pool_id}/providers/${google_iam_workload_identity_pool_provider.provider.workload_identity_pool_provider_id}",
+  "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+  "token_url": "https://sts.googleapis.com/v1/token",
+  "credential_source": {
+    "url": "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=api://${local.app_name}",
+    "headers": {
+      "Metadata": "True"
+    },
+    "format": {
+      "type": "json",
+      "subject_token_field_name": "access_token"
+    }
+  },
+  "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/${module.sa.email}:generateAccessToken"
+}
+EOF
+sudo apt -y update
+sudo apt -y install apt-transport-https ca-certificates gnupg jq
+echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+sudo apt-get update && sudo apt-get install google-cloud-sdk
+CUSTOM_DATA
+}
+
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+data "azuread_client_config" "config" {}
+
+resource "azuread_application" "app" {
```

Comment:
As discussed, I would make Azure resource creation (vm, etc) optional if users have existing resources they want to leverage.

Reply:
done